pkg/container-collection/options.go

```diff
@@ -47,17 +47,38 @@ import (
 
 func enrichContainerWithContainerData(containerData *runtimeclient.ContainerData, container *Container) {
 	// Runtime
-	container.Runtime.ContainerID = containerData.Runtime.ContainerID
-	container.Runtime.RuntimeName = containerData.Runtime.RuntimeName
-	container.Runtime.ContainerName = containerData.Runtime.ContainerName
-	container.Runtime.ContainerImageName = containerData.Runtime.ContainerImageName
-	container.Runtime.ContainerImageDigest = containerData.Runtime.ContainerImageDigest
+	if container.Runtime.ContainerID == "" {
+		container.Runtime.ContainerID = containerData.Runtime.ContainerID
```

Comment:
Can you explain in the commitmsg why we now need to check if each field is empty?
Is enrichContainerWithContainerData() called several times, and from where?

Reply:
Added commit message:

> If the container is not fully enriched, this function will be called
multiple times. So if some RuntimeEnricher do not have all informations
about a given Container, it should not overwrite the current container
informations, only update it